adapter/internal/ga/client.go

```diff
@@ -102,6 +105,26 @@ func initConnection() (*grpc.ClientConn, error) {
 	return conn, nil
 }
 
+func generateTLSCredentialsForXdsClient() credentials.TransportCredentials {
+	conf, _ := config.ReadConfigs()
+	certPool := tlsutils.GetTrustedCertPool(conf.Adapter.Truststore.Location)
+	// There is a single private-public key pair for XDS server initialization, as well as for XDS client authentication
+	certificate, err := tlsutils.GetServerCertificate(conf.Adapter.Keystore.PublicKeyLocation,
```

Comment:
Shouldn't we use two configs for key and certificate?

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2150/commits/13c2a3b94cd52cef14753f079b0af775a2fd417b